datafusion/functions-aggregate/src/variance.rs

```diff
@@ -316,6 +316,7 @@ fn merge(
     mean2: f64,
     m22: f64,
 ) -> (u64, f64, f64) {
+    debug_assert!(count != 0 || count2 != 0, "Cannot merge two empty states");
```

Comment:
I don't exactly know how it works. Is supporting this not needed?

Reply:
It is a precondition of using this function. So support is not needed here, but expected to be handled by callers. When both counts are 0 then also `mean` and `m2` is 0 and the correct result would also be all 0.

Reply:
i should have checked the callers. thanks. 👍